Detect if users attempt to create a dataset

knirirr opened the pull request:
Once this is merged then I can also merge the relevant API code:
https://github.com/FAIRsharing/FAIRsharing-API/pull/219#pullrequestreview-563427689

What this does is set a flag ("is_dataset") which is passed to the API to deal with, if the following user behaviour is encountered:

1. Selecting a "database" of some sort as the type of their new record.
2. Clicking "yes" when asked if they intend to upload a dataset.
3. Clicking "go back" when seeing the message that says they should stop because we don't accept datasets.
4. Selecting "database" again.
5. Clicking "no" when the question comes up again.

At present it will be triggered any time a database record is selected rather than just on create. But, disabling the changing of types other than for curators needs to be done.

knirirr committed:
Client-side code for feature in #577.

knirirr committed:
Autogenerated documentation files

terazus commented on src/components/Editor/GeneralInformation/DatabaseWarning.vue:
```diff
@@ -113,6 +113,13 @@
+          },
+          pressNo(){
+            this.showOverlay = false;
+            if (this.submitted === 1) {
```
```suggestion
            if (this.submitted >= 1) {
```
=> so if a user presses more than once, it also sends the email :)

knirirr committed:
Changed === to >= as per
https://github.com/FAIRsharing/fairsharing.github.io/pull/643#pullrequestreview-563762208

knirirr committed:
Merge branch 'mt_dataset_email_577' of github.com:FAIRsharing/fairsharing.github.io into mt_dataset_email_577

knirirr committed:
Autogenerated documentation files
